firmware/application/external/app_manager/ui_app_manager.cpp

```diff
@@ -0,0 +1,273 @@
+#include "ui_app_manager.hpp"
+#include "ui_navigation.hpp"
+#include "ui_receiver.hpp"
+#include "ui_touch_calibration.hpp"
+#include "ui_text_editor.hpp"
+#include "ui_external_items_menu_loader.hpp"
+
+#include "portapack_persistent_memory.hpp"
+#include "lpc43xx_cpp.hpp"
+using namespace lpc43xx;
+
+#include "audio.hpp"
+#include "portapack.hpp"
+using namespace portapack;
+
+#include "file.hpp"
+#include "file_path.hpp"
+namespace fs = std::filesystem;
+
+#include "string_format.hpp"
+#include "ui_font_fixed_8x16.hpp"
+#include "cpld_update.hpp"
+#include "config_mode.hpp"
+#include "i2cdevmanager.hpp"
+#include "i2cdev_max17055.hpp"
+
+#include "usb_serial_asyncmsg.hpp"
+#include "file_reader.hpp"
+
+
+using namespace portapack;
+
+
+namespace ui::external_app::app_manager {
+
+/* AppManagerView **************************************/
+/* | inner apps | external apps |
+ * | id         | display name  |
+ * | callname   | friendly name |
+ */
+AppManagerView::AppManagerView(NavigationView& nav)
+    : nav_{nav} {
+    add_children({&labels,
+                  &menu_view,
+                  &text_app_info,
+                  &button_hide_unhide,
+                  &button_clean_hide,
+                  &button_set_cancel_autostart,
+                  &button_clean_autostart});
+
+    menu_view.set_parent_rect({0, 2 * 8, screen_width, 24 * 8});
+
+    menu_view.on_highlight = [this]() {
+        if (menu_view.highlighted_index() >= app_list_index) {
+            text_app_info.set("");
+            return;
+        }
+
+        std::string info;
+        auto app_name = get_app_info(menu_view.highlighted_index(), true);
+        auto app_id = get_app_info(menu_view.highlighted_index(), false);
+
+        info += "Hidden:";
+
+        if (is_blacklisted(app_name)) {
+            info += "Yes ";
+        } else {
+            info += "No ";
+        }
+
+        info += "Autostart:";
+        if (is_autostart_app(app_id)) {
+            info += "Yes";
+        } else {
+            info += "No";
+        }
+
+        if (info.empty()) {
+            info = "Highlight an app";
+        }
+
+        text_app_info.set(info);
+    };
+
+    button_hide_unhide.on_select = [this](Button&) {
+        hide_unhide_app();
+        refresh_list();
+    };
+
+    button_clean_hide.on_select = [this](Button&) {
+        clean_blacklist();
+        refresh_list();
+    };
+
+    button_set_cancel_autostart.on_select = [this](Button&) {
+        set_unset_autostart_app();
+        refresh_list();
+    };
+
+    button_clean_autostart.on_select = [this](Button&) {
+        unset_auto_start();
+        refresh_list();
+    };
+
+    refresh_list();
+}
+
+void AppManagerView::refresh_list() {
+    auto previous_cursor_index = menu_view.highlighted_index();
+    app_list_index = 0;
+    menu_view.clear();
+
+    size_t index = 0;
+
+    auto padding = [](const std::string& str) {
+        return str.length() < 25 ? std::string(25 - str.length(), ' ') + '' : "";
```

Comment:
github diff view doesn't show that char, but portapack can show it

Reply:
Okay

Reply:
Okay